modules/authentication/proc-enabling-authentication-with-github.adoc

```diff
@@ -0,0 +1,149 @@
+[id="enabling-authentication-with-github"]
+= Enabling authentication with GitHub
+
+To authenticate users with GitHub, enable the GitHub authentication provider in {product}.
+
+.Prerequisites
+* You link:https://docs.redhat.com/en/documentation/red_hat_developer_hub/{product-version}/html/administration_guide_for_red_hat_developer_hub/assembly-add-custom-app-file-openshift_admin-rhdh[added a custom {product-short} application configuration], and have sufficient permissions to modify it.
+* You have sufficient permissions in GitHub to create and manage a link:https://docs.github.com/en/apps/overview[GitHub App].
+
+.Procedure
+. To allow {product-short} to authenticate with GitHub, create a GitHub App.
+Prefer a GitHub App to an OAuth app to use fine-grained permissions, give more control over which repositories the application can access, and use short-lived tokens.
```

Comment:
```suggestion
Opt for a GitHub App instead of an OAuth app to use fine-grained permissions, gain more control over which repositories the application can access, and use short-lived tokens.
```